### PR TITLE
Add mailbox command parameter validity checking

### LINKED ISSUE #787: MB Command Parameter Checking for Run-Time
All Input Data must have parameter checking.  This mostly affects our MB implementation.

For example, if a size value is passed in with the MB command, we should verify that size value against max values and other bounds (like message ptr increments) to ensure that attackers can't attack via MB input.

### PR DESCRIPTION
**Pull request by sree-revoori1**
For invoke DPE, we check that data_size <= MAX_DATA_SIZE.

Fixes #787

### CODE REVIEW

File: runtime/src/invoke_dpe.rs [hunk 1]
```diff
@@ -10,6 +10,14 @@ pub struct InvokeDpeCmd;
 impl InvokeDpeCmd {
     pub(crate) fn execute(drivers: &mut Drivers, cmd_args: &[u8]) -> CaliptraResult<MailboxResp> {
         if let Some(cmd) = InvokeDpeReq::read_from(cmd_args) {
+            // Validate data length
+            let Ok(in_len) = usize::try_from(cmd.data_size) else {
```
Comment (jhand2) [inline runtime/src/invoke_dpe.rs:14]: nit: Caliptra firmware won't ever run on a processor that is less than 32-bit, so `cmd.data_size as usize` is safe.

Comment (jhand2) [inline runtime/src/invoke_dpe.rs:14]: It looks like we actually don't use `cmd.data_size`. I think we need to update the execute call to `.execute_serialized_command(&mut env, drivers.mbox.user(), &cmd.data[..cmd.data_size as usize])`

Comment (sree-revoori1) [reply runtime/src/invoke_dpe.rs:14]: Done

Comment (sree-revoori1) [reply runtime/src/invoke_dpe.rs:14]: Done.

**Review by jhand2 — DISMISSED** [review]

**Review by jhand2 — APPROVED** [review]

**Pull request merged**